src/minesweeper.rs

```diff
@@ -212,33 +212,33 @@ impl Minesweeper {
                     let state = self.mine_states[index].cycle();
                     self.mine_states[index] = state;
                     visual.set_brush(self.get_color_brush_from_mine_state(state))?;
-                } else if self.mine_states[index] == MineState::Empty {
-                    if self.sweep(self.current_selection_x, self.current_selection_y)? {
-                        // We hit a mine! Setup and play an animation whiel locking any input.
-                        let hit_x = self.current_selection_x;
-                        let hit_y = self.current_selection_y;
-
-                        // First, hide the selection visual and reset the selection
-                        self.selection_visual.set_is_visible(false)?;
-                        self.current_selection_x = -1;
-                        self.current_selection_y = -1;
-
-                        // Create an animation batch so that we can know when the animations complete
-                        let batch = self
-                            .compositor
-                            .create_scoped_batch(CompositionBatchTypes::Animation)?;
-
-                        self.play_animation_on_all_mines(hit_x, hit_y)?;
-
-                        // Subscribe to the completion event and complete the batch
-                        // TODO: events
-                        batch.end()?;
-
-                        self.mine_animation_playing = true;
-                        self.game_over = true;
-                    }
-                    // TODO: Detect that the player has won
+                } else if self.mine_states[index] == MineState::Empty
+                    && self.sweep(self.current_selection_x, self.current_selection_y)?
```

Comment:
Hmmmm, clippy is technically "correct" here, but it ignores the intent of the TODO. My intention was that the separate if clause would have its own else to determine if the user had won. When I implement that feature I'm going to have to separate this back out anyway. Could we undo just this part? Otherwise the rest of the change looks righteous to me.